tests/testthat/test-5dominance.R

```diff
@@ -0,0 +1,29 @@
+context("estimateDominance")
+
+test_that("estimateDominance", {
```

Comment:
Unit test should be structured slighlty differently. The main focus is to make sure it work as intended, not that it works.

Two way exists:
- Either you compute it by two different ways
- you compare the results to fixed value, you know to be true.

The examples below show that it works but not necessarily as intended. In the follwoing code chunk is an example, which you can use as blue print for adding all the tests.
```
# .simpson_dominance
x <- 1
expect_equal(mia:::.simpson_dominance(x),1)
x <- 1:1000
expect_equal(mia:::.simpson_dominance(x), 0.001332667)
```

So in this style you need to make sure that the follwing functions work correctly:
- `.gini_dominance`
- `.get_core_dominance`
- `.get_dominance` for all four cases

then you check if all indeces get calculated for the call `estimateDominance(esophagus)` by checking the colnames of `colData()` after making sure, that the return value of `estimateDominance` is actually the same type as the input.

Reply:
Ok, thanks for help. I changed tests little bit. All, but 3 are working. It seems that .simpson_dominance should return value 1-a,  not a.

Reply:
Then you have uncovered and error, which you can fix in the code and the tests.

Reply:
And I really like that you test against vegan functions. You can fix the error in the code of simpson_dominance, if you are sure about it. @antagomir

Reply:
Simpson is confusing bc there are [different indices](https://en.wikipedia.org/wiki/Diversity_index#Simpson_index).

The _Simpson index_ is defined as $\lambda = \Sum_i p^2_i$. 

Simpson dominance index is defined as $\lambda$. It is the same as Simpson index (one source [here](https://www.sciencedirect.com/science/article/pii/S2405844019362668)

Inverse Simpson diversity index is defined as $\frac{1}{\lambda}$.

Then we also have Gini-Simpson index $1 - \lambda$.

Would be nice to have all of these in the package but the definitions must be clear in order to avoid misuse. Just check that the function corresponds to these official definitions (inverse Simpson and Gini-Simpson should go to `estimateDiversity` whereas Simpson should go to `estimateDominance`).

Another point to check that all dominance indices (and all diversity indices) have the same sign. In some cases the interpretation of alpha diversity/dominance indices can be inverse compared to the other indices. Where this is the case, we may like to consider how to treat. It is causing some risks for interpretation if the indices with the same purpose have opposite interpretations for increasing index values (higher values meaning either decreasing or increasing diversity or dominance). Could you check on the same go if there are such cases so that we can see what to do. That discussion can go in a new issue.

Reply:
Modified simpson tests. I calculated simpson index with pen and paper and tested .simpson_dominance with that